From 6256e5b884d78c33d94598f4ed93f747dc5f1a22 Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Mon, 11 Aug 2014 16:51:31 +0300
Subject: [PATCH] telcomm: handle null subscription number

BZ: 216483

It is possible that the subscription number is unavailable.
Handle this case by checking for null before attempting to
serialize it to telcomm database

Change-Id: Iaedb293222d9cf0f1fc46de2fd164d5bb089c488
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../android/telecomm/PhoneAccountRegistrar.java    |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/com/android/telecomm/PhoneAccountRegistrar.java b/src/com/android/telecomm/PhoneAccountRegistrar.java
index f6e522d..fb18bc4 100644
--- a/src/com/android/telecomm/PhoneAccountRegistrar.java
+++ b/src/com/android/telecomm/PhoneAccountRegistrar.java
@@ -497,7 +497,9 @@ public final class PhoneAccountRegistrar {
             serializer.endTag(null, HANDLE);
 
             serializer.startTag(null, SUBSCRIPTION_NUMBER);
-            serializer.text(o.getSubscriptionNumber());
+            if (o.getSubscriptionNumber() != null) {
+                serializer.text(o.getSubscriptionNumber());
+            }
             serializer.endTag(null, SUBSCRIPTION_NUMBER);
 
             serializer.startTag(null, CAPABILITIES);
-- 
1.7.9.5

